sale_order_global_stock_route/models/sale_order.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2020 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    route_id = fields.Many2one(
+        comodel_name="stock.location.route",
+        string="Route",
+        domain=[("sale_selectable", "=", True)],
+        help="When you change this field all the lines will be changed."
+        " After use it you will be able to change each line.",
+    )
+
+    @api.onchange("route_id")
+    def _onchange_route_id(self):
+        """We could do sale order line route_id field compute store writable.
+        But this field is created by Odoo so I prefer not modify it.
+        """
+        for line in self.order_line:
+            line.route_id = line.order_id.route_id
+
+    def write(self, vals):
+        res = super().write(vals)
+        if "route_id" in vals:
+            lines = self.mapped("order_line").filtered(
+                lambda l: l.route_id.id != vals["route_id"]
+            )
+            lines.write({"route_id": vals["route_id"]})
+        return res
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
```

Comment:
please put one model per file

Reply:
With one2many relations our guidelines are to keep the model along with the parent model stuff. Please reverse this change

Reply:
Understood